[Weng Ying] iP

wengYing227 opened the pull request:
# Duke
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))
<!-- blank line -->
Duke helps you reminder of the tasks you need to do.
Cons are as follows:
- Text-based
- Easy-to-go
- convenient
<!-- blank line -->
All you need to do is only:
1. download it from [the link](https://github.com/nus-cs2103-AY2122S1/ip) and open it.
2. add your tasks.
3. finish your tasks.
4. any other operations like _showing the whole list_, etc. :star_struck:
<!-- blank line -->
The most important thing is - the app is all **FREE**!!!
<!-- blank line -->
Features are:
- [x] Managing Todo
- [x] Managing Deadline
- [x] Managing Event
- [ ] Reminder (coming soon~~~)
- [ ] ~~Other features I hadn't thought of yet~~

----

If you are a Java programmer, you can try this to practice your Java skills. 
Here is an example of the `main` method:
```java
public static void main(String[] args) {
        DukeDB data = new DukeDB("data/tasks.txt");
        data.addData(new Todo("hahaha"));
        ArrayList<Task> lst = data.readData();
        for (int i = 0 ; i < lst.size(); i++) {
            System.out.println(lst.get(i));
        }
        data.deleteData(1);
        data.doneData(0);
}
```

Commit:
Add Gradle support

wengYing227 committed:
Greet, Echo, Exit

wengYing227 committed:
Add, List

wengYing227 committed:
Mark as Done

wengYing227 committed:
ToDos, Events, Deadlines

wengYing227 committed:
Automated Text UI Testing

wengYing227 committed:
Handle Errors

wengYing227 committed:
Delete

wengYing227 committed:
enum

wengYing227 committed:
Adds database

wengYing227 committed:
Adds Date and Time

wengYing227 committed:
Merge branch 'branch-Level-8'

wengYing227 committed:
Uses more OOP

wengYing227 committed:
Organises into packages

wengYing227 committed:
Adds JUnit Tests

wengYing227 committed:
Creates a JAR File

wengYing227 committed:
Adds JavaDoc

wengYing227 committed:
Follows the Coding Standard

wengYing227 committed:
Follows the Coding Standard

wengYing227 committed:
Adds Find

wengYing227 committed:
Merge branch 'branch-A-JavaDoc' into branch-A-CodingStandard

wengYing227 committed:
Merge branch 'branch-Level-9' into branch-A-CodingStandard

# Conflicts:
#	src/main/java/duke/util/Parser.java
#	src/main/java/duke/util/Ui.java

wpinrui commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,44 @@
+    }
+
+    /**
+     * Start the program
```
Should this be 'Starts' instead of 'Start'?

wpinrui commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,69 @@
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a deadline task with a <code>description</code> corresponding to the content
```
Why are there html tags \<code> surrounding some of the words in the JavaDoc?

wpinrui commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,55 @@
+package duke.task;
```
LGTM

wpinrui commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,16 @@
+package duke.task;
```
LGTM

wpinrui commented on src/main/java/duke/util/DukeDB.java:
```diff
@@ -0,0 +1,226 @@
+     * Parses a full command into a Task object,
+     * example of command should be like: E & 0 & project meeting & 6/8/2021 1400 .
+     *
+     * @param string the original full command
```
Should the 'the' be capitalised?

wpinrui commented on src/main/java/duke/util/DukeException.java:
```diff
@@ -0,0 +1,10 @@
+package duke.util;
```
LGTM

wpinrui commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,140 @@
+    }
+
+    /**
+     * Parses an input and process it,
```
Perhaps the readability of the JavaDoc could be improved?

wpinrui commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,140 @@
+                throw new DukeException("OOPS!!! The timeline of a event cannot be empty.");
+            }
+            String item = input.substring(6, indexOfTime);
+            String at = input.substring(indexOfTime+4);
```
Should this be `indexOfTime + 4` with the spacing?

wpinrui commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,140 @@
+                taskList.add(todo);
+                storage.add(todo);
+                ui.showNewTask(todo);
+            } catch(StringIndexOutOfBoundsException e) {
```
Should there be a whitespace after the keyword `catch`?

wpinrui commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,140 @@
+        if (taskList.length() > 0) {
+            items = items + "1. " + taskList.get(0).toString();
+        }
+        for (int i = 2 ; i <= taskList.length(); i++) {
```
Should the semi-colon be right after the 2?

wpinrui commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,140 @@
+        return command.toLowerCase().equals("bye");
+    }
+
+    public void findKeyword(String keyword) {
```
Would it be good to have a JavaDoc for this method?

wengYing227 replied:
> Would it be good to have a JavaDoc for this method?

Thanks for pointing out those non-standard coding style that I didn't realize!

wpinrui commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,64 @@
+package duke.util;
```
LGTM

wpinrui commented on src/main/java/duke/util/Ui.java:
```diff
@@ -0,0 +1,110 @@
+ * Represents the User Interfacer of Duke.
+ */
+public class Ui {
+    public void print(String... strings) {
```
I like the use of varargs :smiley:

wpinrui reviewed (commented):
Overall, I did not find many coding standard violations apart from a few whitespace issues. The indentation seemed to be very good.

wengYing227 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

wengYing227 committed:
Uses Gradle

wengYing227 committed:
Uses CheckStyle

itsraveen commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,79 @@
+        return by;
+    }
+
+    /**
+     * Parses a time String into a LocalDateTime.
+     *
+     * @param time a String of format dd/mm/yyyy hhmm(optional)
+     * @return transformed time of type LocalDateTime
+     */
```
JavaDocs very clear by indicating the format for the date.

itsraveen commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,151 @@
+     *
+     * @param input a full command
+     */
+    public void parse(String input) {
```
Perhaps extract out the methods for each if else block to make the current method shorter and more readable?

itsraveen commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,65 @@
+import java.util.ArrayList;
+
+import duke.task.Task;
+
+
+
```
Small issue, but can remove these unnecessary extra lines! 😄

itsraveen reviewed (commented):
Generally looks good to me. Code follows the coding standard and is easy to read and understand.

wengYing227 committed:
Looking for error

wengYing227 committed:
Remains error

wengYing227 committed:
Unfinished

wengYing227 committed:
Adds GUI

wengYing227 committed:
Revert "Unfinished"

This reverts commit 3a01ca359361569b4826ea8608c7518f1da879a6.

wengYing227 committed:
Revert "Revert "Unfinished""

This reverts commit a367a1c6b312ade66029f015069abc660dd9963f.

wengYing227 committed:
Merge branch 'error'

# Conflicts:
#	build.gradle
#	data/tasks.txt
#	src/main/java/duke/Duke.java
#	src/main/java/duke/util/Parser.java
#	src/main/java/duke/util/Ui.java

wengYing227 committed:
Fix violation of code quality

wengYing227 committed:
Use Assertions

There may be some parts where the programmer(i.e. me) make wrong assumptions.

Let's insert assert expressions to some important points like in the Parser class.

Using assert feature to document these important assumptions in the code allows us to discover those wrong assumptions.

Refers to: https://nus-cs2103-ay2122s1.github.io/website/schedule/week5/topics.html#w5-6-assertions

wengYing227 committed:
Improve code quality

Code with high quality is readable, understandable and can be modified easily by other programers.

Avoiding deep nesting, long methods, complicated expressions, magic numbers, etc.
Trying to refactor so that the code is logical and simple.

Following those code quality conventions helps to make our code more readable.

Refers to: https://nus-cs2103-ay2122s1.github.io/website/schedule/week5/topics.html#w5-4-code-quality

wengYing227 committed:
Merge pull request #2 from wengYing227/branch-A-Assertions

Use Assertions

wengYing227 committed:
Merge branch 'master' into branch-A-CodeQuality

wengYing227 committed:
Merge pull request #3 from wengYing227/branch-A-CodeQuality

Improve code quality

wengYing227 committed:
Add CI

GitHub Actions is CI/CD tool integrated into GitHub.

Use GitHub Actions to set up Continuous Integration (CI).

Refers to: https://se-education.org/guides/tutorials/githubActions.html

wengYing227 committed:
Add Priority to items

Provide an attribute to each item so that we can attach any one of three priority levels(high/medium/low).

Priority level attachment allows users to tag urgent/important task items.

Adding a new attribute 'priority' and corresponding methods.

wengYing227 committed:
Merge pull request #4 from wengYing227/branch-BCD-Extension

Add Priority to items (C-Priority)

wengYing227 committed:
Adds Level-10

wengYing227 committed:
Improve GUI

The graphical user inteface looks simpler and better.

The GUI of the original version looks like something from the '90s.
Users might refuse to use it because of the bad GUI design.

Modifying the design of dialog boxes, the size of the whole window,
and the font of 'Send' button.

Changing the parts can make the whole GUI look simple and modern.

Refer to seniors' work
https://nus-cs2103-ay2021s1.github.io/website/admin/ip-showcase.html

wengYing227 committed:
Improves GUI

Better GUI can enhance good user experience.

Adding exit command and modifying response format.

wengYing227 committed:
Add Ui image

wengYing227 committed:
Add User Guide

User Guide helps user to learn how to use the app efficiently.

Adding a README file as User Guide.

wengYing227 committed:
Update README.md

wengYing227 committed:
Update README.md

wengYing227 committed:
Update README.md

wengYing227 committed:
Update README.md

wengYing227 committed:
Add more test cases

wengYing227 committed:
Modify README file in the homepage of github

wengYing227 committed:
Update README.md

wengYing227 committed:
Modify output

wengYing227 committed:
Merge branch 'master' of https://github.com/wengYing227/ip

wengYing227 committed:
Fix bugs on Database

wengYing227 committed:
Fix bugs on invalid inputs

wengYing227 committed:
Improves GUI design

wengYing227 committed:
Improve GUI
